### PR TITLE
[docs] Go live with the new urls!

### PR DESCRIPTION
**Pull request by siriwatknp**
**Deploy Preview**: https://deploy-preview-4313--material-ui-x.netlify.app/x/react-data-grid/
**Migration subdomain**: https://migration.mui.com/x/react-data-grid/

**Changes**
- in dev, redirect from `/` to `/x/data-grid/`
- update all the links in the code base to the new one. eg. `/components/x/react-date-pickers/*` to `/x/react-date-pickers/*`
- remove old pages

**Checklist**
- [x] Run the ahrefs audit
- [x] Update monorepo to latest (need to merge https://github.com/mui/material-ui/pull/32048 first)
- [x] Change date pickers navigation icon (need to be done in the Core repo)
- [x] Date pickers product identifier
  Screen Shot 2565-04-01 at 08 45 28

> the production redirects will be set on the Core repo

### CODE REVIEW

**Comment by siriwatknp** [comment]
@flaviendelangle I found several files still referencing to this URL.

```js
  /**
   * ...
   * [Read the guide](https://next.material-ui-pickers.dev/guides/forms) about form integration and error displaying.
   * @DateIOType
   *
   * @param {TError} reason The reason why the current value is not valid.
   * @param {TDateValue} value The invalid value.
   */
  onError: PropTypes.func,
```

What should be the correct one for `https://next.material-ui-pickers.dev/guides/forms`?

**Comment by flaviendelangle** [comment]
@siriwatknp no page talks about `Formik` or `integration` so I think this page never had an equivalent on the lab and therefore no equivalent on X.

Can we let it as is and create an issue to add a clean page in our X doc later ?

**Comment by siriwatknp** [comment]
> @siriwatknp no page talks about `Formik` or `integration` so I think this page never had an equivalent on the lab and therefore no equivalent on X.
> 
> Can we let it as is and create an issue to add a clean page in our X doc later ?

Sure, no problem. I will leave it as is.

**Comment by flaviendelangle** [comment]
https://github.com/mui/mui-x/issues/4320

**Comment by cherniavskii** [comment]
Should we consider adding redirect rules for the pages that were removed?
My main concern is that links to old pages in Github issues will be broken

**Comment by cherniavskii** [comment]
Was there a discussion around versions page?
Currently, "View all versions" button links to https://mui.com/versions/
Screenshot 2022-04-06 at 11 55 22

It doesn't consider Advanced components versions
Screenshot 2022-04-06 at 11 55 31

Can we remove this link for Advanced components until we come up with a better solution? Ideally we should have separate versioning page(s) for Advanced components

**Comment by flaviendelangle** [comment]
> Ideally we should have separate versioning page(s) for Advanced components

Or a single page with the compatibility table between X and Core if we desync the versions

**Comment by siriwatknp** [comment]
> Should we consider adding redirect rules for the pages that were removed? My main concern is that links to old pages in Github issues will be broken
> 
> UPDATE: I just noticed that those redirects are present in Migration subdomain, so I assume this is what we'll see in production as well?

Correct, all the redirects including data-grid and date-pickers have been done on the Core repo. Please have a look at https://github.com/mui/material-ui/pull/32048/files#diff-7a0cbac74402caf0e99ba338a9e7c42684580a1f85e149110c3c2ca6e38055af.

**Comment by siriwatknp** [comment]
> Was there a discussion around versions page?
> Currently, "View all versions" button links to https://mui.com/versions/

Good point, I think I will remove the "All versions" from the data grid for now.

**Review by cherniavskii — APPROVED** [review]

File: test/e2e-website/data-grid.spec.ts [hunk 1]
```diff
@@ -42,9 +42,10 @@ test.describe('DataGrid docs', () => {
     test('should have correct link for sidebar anchor', async ({ page }) => {
       await page.goto(`/x/react-data-grid/`);
 
-      const anchor = page.locator('nav[aria-label="documentation"] ul ul a:text-is("Overview")');
+      const anchor = page.locator('nav[aria-label="documentation"] ul ul a:text-is("Data Grid")');
```
Comment (cherniavskii) [inline test/e2e-website/data-grid.spec.ts:45]: Why "Data Grid"?

Comment (siriwatknp) [reply test/e2e-website/data-grid.spec.ts]: This was wrong. I have reverted it.

**Pull request merged**